src/test/java/de/tum/in/www1/artemis/domain/notification/GroupNotificationFactoryTest.java

```diff
@@ -0,0 +1,408 @@
+    }
+
+    private void checkCreatedNotificationWithoutNotificationText() {
+        checkCreatedNotification(createdNotification, expectedTitle, expectedText, expectedTarget, expectedPriority);
+    }
+
+    private String createDefaultExpectedTarget(String message, String entity, Long relevantIdForCurrentTestCase) {
+        return "{\"message\":\"" + message + "\",\"id\":" + relevantIdForCurrentTestCase + ",\"entity\":\"" + entity + "\",\"course\":" + courseId + ",\"mainPage\":\"courses\"}";
+    }
+
+    private enum Base {
+        ATTACHMENT, EXERCISE, POST, ANSWER_POST, COURSE, EXAM
+    }
+
+    /**
+     *  Calls the real createNotification method of the groupNotificationFactory and tests if the result is correct.
+     * Two notifications are created for those cases that might use a manually set notification text
+     * @param base is the first input parameter used in the respective factory method to create the group notification.
+     */
+    private void createAndCheckNotification(Base base) {
+        switch (base) {
+            case ATTACHMENT: {
+                createdNotification = groupNotificationFactory.createNotification(attachment, user, groupNotificationType, notificationType, notificationText);
+                checkCreatedNotificationWithNotificationText();
+                createdNotification = groupNotificationFactory.createNotification(attachment, user, groupNotificationType, notificationType, null);
+                break;
+            }
+            case EXERCISE: {
+                createdNotification = groupNotificationFactory.createNotification(exercise, user, groupNotificationType, notificationType, notificationText);
+                checkCreatedNotificationWithNotificationText();
+                createdNotification = groupNotificationFactory.createNotification(exercise, user, groupNotificationType, notificationType, null);
+                break;
+            }
+            case POST: {
+                createdNotification = groupNotificationFactory.createNotification(post, user, groupNotificationType, notificationType);
+                break;
+            }
+            case ANSWER_POST: {
+                createdNotification = groupNotificationFactory.createNotification(answerPost, user, groupNotificationType, notificationType);
+                break;
+            }
+            case COURSE: {
+                createdNotification = groupNotificationFactory.createNotification(course, user, groupNotificationType, notificationType, archiveErrors);
+                break;
+            }
+            case EXAM: {
+                createdNotification = groupNotificationFactory.createNotification(exam, user, groupNotificationType, notificationType, archiveErrors);
+                break;
+            }
+        }
+        checkCreatedNotificationWithoutNotificationText();
+    }
+
+    // Based on Attachment
+
+    @Test
+    public void createNotificationBasedOnAttachment() {
+
```

Comment:
```suggestion
```
Don't want to be spamming the review with empty lines 😄  , but there's plenty of other cases in this file where this extra line is present at the beginning of the function. Unless this is a pattern I am unaware of maybe it would be better to remove this in the rest of the file?

Reply:
Is removed 👍 (I just prefer the look of an empty starting line sometimes, but yeah this does add one useless LOC)